presto-spi/src/main/java/com/facebook/presto/spi/predicate/NotExpression.java

```diff
@@ -62,6 +62,16 @@ public TupleExpression transform(Function function)
         return visitor.visitNotExpression(this, context);
     }
 
+    public boolean isNone()
+    {
+        return expression instanceof AllExpression || expression.isAll();
+    }
+
+    public boolean isAll()
+    {
+        return expression instanceof NoneExpression || expression.isNone();
```

Comment:
None ensures that no value is selected (No value is selected in a stripe). So Not(None) = all...(All ensures that all values are selected (Select the entire stripes).